doc/src/tutorial/solvers.rst

```diff
@@ -23,89 +23,104 @@ However, there is an even easier way.  In SymPy, any expression not in an
 = b` if and only if `a - b = 0`, this means that instead of using ``x == y``,
 you can just use ``x - y``.  For example
 
-    >>> solve(Eq(x**2, 1), x)
-    [-1, 1]
-    >>> solve(Eq(x**2 - 1, 0), x)
-    [-1, 1]
-    >>> solve(x**2 - 1, x)
-    [-1, 1]
+    >>> solveset(Eq(x**2, 1), x)
+    {-1, 1}
+    >>> solveset(Eq(x**2 - 1, 0), x)
+    {-1, 1}
+    >>> solveset(x**2 - 1, x)
+    {-1, 1}
 
 This is particularly useful if the equation you wish to solve is already equal
-to 0.  Instead of typing ``solve(Eq(expr, 0), x)``, you can just use
-``solve(expr, x)``.
+to 0. Instead of typing ``solveset(Eq(expr, 0), x)``, you can just use
+``solveset(expr, x)``.
 
 Solving Equations Algebraically
 ===============================
 
-The main function for solving algebraic equations, as we saw above, is
-``solve``.  The syntax is ``solve(equations, variables)``, where, as we saw
-above, ``equations`` may be in the form of ``Eq`` instances or expressions
+The main function for solving algebraic equations is ``solveset``.
+The syntax for ``solveset`` is ``solveset(equation, variable=None, domain=S.Complexes)``
+Where ``equations`` may be in the form of ``Eq`` instances or expressions
 that are assumed to be equal to zero.
 
-.. TODO: This is a mess, because solve() has such a complicated interface.
+Please note that there is an another function called as ``solve`` which
+can also be used to solve equations. The syntax is ``solve(equations, variables)``
+However, it is recommended to use ``solveset`` instead.
 
-When solving a single equation, the output of ``solve`` is a list of the
-solutions.
+When solving a single equation, the output of ``solveset`` is a ``FiniteSet`` or
+an ``Interval`` or ``ImageSet`` of the solutions.
 
-    >>> solve(x**2 - x, x)
-    [0, 1]
+    >>> solveset(x**2 - x, x)
+    {0, 1}
+    >>> solveset(x - x, x, domain=S.Reals)
+    ℝ
+    >>> solveset(sin(x) - 1, x, domain=S.Reals)
+    ⎧        π        ⎫
+    ⎨2⋅n⋅π + ─ | n ∊ ℤ⎬
+    ⎩        2        ⎭
 
-If no solutions are found, an empty list is returned, or
-``NotImplementedError`` is raised.
 
-    >>> solve(exp(x), x)
-    []
+If there are no solutions, an ``EmptySet`` is returned and if it
+is not able to find solutions then a ``ConditionSet`` is returned.
 
-.. note::
+    >>> solveset(exp(x), x)     # No solution exists
+    ∅
+    >>> solveset(Abs(x) - 1, x) # Not able to find solution
```

Comment:
Isn't `cos(x) = x` already used somewhere in this tutorial as an example of an equation with solutions that can't be represented in closed form? I would re-use that.